app/Http/Controllers/PollController.php

```diff
@@ -30,12 +34,26 @@ public function store(Request $request)
     {
         $poll = Poll::create([
             'name' => $request->name,
-            'url' => $request->url,
-          ]);
-        attach($poll->users(), $request->users);
+            'url' => ""
+        ]);
+        attach($poll->users(), $request->users, "admin");
         attach($poll->items(), $request->items);
-    
-        return new PollResource(Poll::with('users', 'items')->find($poll->id));
+
+        return response()->json();
+    }
+
+    /**
+     * Store a newly created rating in storage in the poll_ratings table.
+     *
+     * @param  \Illuminate\Http\Request  $request$
+     * @param  int  $pollId
+     * @return \Illuminate\Http\Response
+     */
+    public function rate(Request $request, int $pollId)
+    {
+        foreach (json_decode($request->ratings, true) as $rating) {
+            DB::insert('insert into poll_ratings (poll_id, user_id, item_id, rating) values (?, ?, ?, ?)', [$pollId, Auth::id(), $rating['id'], $rating['rating']]);
```

Comment:
Users can send a rating of 10 billions and the website will allow it, breaking the poll. (We need a validator!)

Reply:
Users can vote in polls they are not belonging to, thus rigging the results for other groups!